src/Tools/BuildValidator/Records.cs

```diff
@@ -27,4 +28,12 @@ internal sealed record AssemblyInfo(string FilePath, Guid Mvid)
         bool Debug,
         string DebugPath);
 
+    internal record ResolvedSource(
```

Comment:
I think I'm getting a little lost in the usage of these records:

- ResolvedSource
- SourceFileInfo
- SyntaxTreeInfo

Could you please clarify why it's necessary to have all 3?

Reply:
At the `BuildConstructor` level there is a need to essentially to pass down a tuple of `(string FilePath, SourceText Text)` to the language specific routine so that it can generate a `SyntaxTree`. 

Before my change `ResolvedSource` served that purpose because we were only driving this operation from the `BuildValidator` tool. The `ResolvedSource` type is used for resolving source locations on disk, it contains the information necessary for creating `SyntaxTree` in `BuildConstructor` and hence it was used. 

From the library / unit test layer though the `ResolvedSource` type contains a lot of unnecessary information for generating `SyntaxTree`. The caller needs to think about items like embedding, hashes, etc ... None of which is actually necessary to generate the `SyntaxTree`. I toyed for a bit with a pattern of having a method called `CreateSyntaxTree(string filePath, SourceText sourceText)` on `BuildConstructor` but that didn't feel right at this time. May be right later though when we get further into the abstractions.

Hence I decided to simplify and introduce `SyntaxTreeInfo` for the moment. I think it makes it clearer what the actual dependencies are for this layer. 

As for why we need `ResolvedSource` and `SourceFileInfo` it's because one specifies what to retrieve and the other what was retrieved. It's possible we could get to a point where we don't need one of them but at the moment there didn't seem to be a clear way.